File: python3/test/doubles/TestShipmentService.py
from src.domain.Order import Order
from src.service.ShipmentService import ShipmentService


class TestShipmentService(ShipmentService):
    def __init__(self):
        self.shipped_order = None

    def ship(self, order: Order):
        self.shipped_order = order

    def get_shipped_order(self):
        return self.shipped_order

File: typescript/test/doubles/TestShipmentService.ts
import Order from '../../src/domain/Order';
import { ShipmentService } from '../../src/service/ShipmentService';

class TestShipmentService implements ShipmentService {
  private shippedOrder: Order = null;

  public getShippedOrder(): Order {
    return this.shippedOrder;
  }

  public ship(order: Order): void {
    this.shippedOrder = order;
  }
}

export default TestShipmentService;
